HCC RT Header needs to trigger re-build

aaronenyeshi opened the pull request:
This resolves JIRA Ticket SWDEV-123229 - HCC RT header changes don't trigger a re-build.

There was an issue with HCC RT Headers not triggering a re-build when running make. This was due to dependencies not being properly set and checked for headers in hcc/include.

aaronenyeshi committed:
Merge pull request #1 from RadeonOpenCompute/clang_tot_upgrade

Merge to recent 844cf36

aaronenyeshi committed:
SWDEV-123229 HCC RT header needs to trigger re-build

SWDEV-123229 - HCC RT header changes don't trigger a re-build.

When calling "make" to rebuild hcc after modifying the HCC RT headers, the build is not able to detect the change and to rebuild the .cpp files that include the headers being modified.  As a result, developers may be getting an hcc built from stale objects.

aaronenyeshi commented:
I have tested this on unit tests, and there are no new testcase failures with this. With this fix, developers can modify headers such as kalmar_runtime.h and run make which will be triggered from header file changes. 

Please let me know if I should also make this change for files include/coordinate, include/array_view, and subdirectory include/experimental/.

whchung commented:
@aaronenyeshi thanks for this PR! please help amend it for other headers within `include` directory. Thanks.

scchan commented on include/CMakeLists.txt:
```diff
@@ -1,13 +1,60 @@
+  hc_norm_unorm.inl
+  hc_short_vector.inl
+  kalmar_short_vectors.inl)
+
```
is there a way to automatically generate these lists instead of constructing them manually?

aaronenyeshi replied:
I will take a look

aaronenyeshi committed:
Allows us to generate dependency without explicitly expressing names

Generate the dependency without having to specify each header files explicitly.

aaronenyeshi reviewed (commented):
Just added another commit to generate the dependency without having to specify each header files explicitly, using regular expressions on the file names.

I still need to add another commit for other header files in the `include `directory.

aaronenyeshi committed:
Modify CMake Header files to trigger re-build

I've added the other header files in the include sub-directories. I've also created a target specific for pstl-headers.

aaronenyeshi commented:
Okay, I've made it generate lists automatically for the header files.
In addition, I've included the other headers inside `include` .
Also, now there is a target for hcc-headers, and pstl-headers.
 
These commits does not produce any new testcase failures. Please take a look. Thanks!

whchung reviewed (approved):
LGTM. @scchan and @pfultz2 ?

scchan commented:
looks like the build failed to generate the .deb package.
@aaronenyeshi could you do a "make package" and see if you could reproduce it?

aaronenyeshi committed:
Fix Jenkins build errors for PR358

Fix Jenkin build errors. Seems it dropped the impl/ sub-directory name in cmake_install.cmake.

aaronenyeshi commented:
@scchan Yes, I was able to find the bug, and fix it. Seems it dropped the sub-directory impl/ string in the cmake-install.cmake built file.

aaronenyeshi commented on include/CMakeLists.txt:
```diff
@@ -1,20 +1,43 @@
+# Create target for hcc-headers and set dependencies
+add_custom_target(hcc-headers ALL DEPENDS ${out_files})
+add_dependencies(world hcc-headers)
+set_target_properties(hcc-headers PROPERTIES FOLDER "HCC Misc")
```
This line (35) might not be useful, as its meant for MS Visual Studio UI navigation. Should I remove it?

whchung replied:
@aaronenyeshi thanks for spotting this. let's remove it for now.

aaronenyeshi replied:
Great! I think it is ready to be merged.

aaronenyeshi committed:
Remove not used MSVC UI navigation lines

Not using Visual Studio UI Navigation here.
